src/sunsynk/definitions3phhv.py

```diff
@@ -399,8 +426,140 @@
+    SelectRWSensor(222, "Battery 1 BMS other flag - Battery 1 Sleep", options={0: "off", 1: "on"}, bitmask=0x08),
+    Sensor(223, "Battery 1 BMS type", ""),
+    Sensor(224, "Battery 1 BMS SOH", "%"),
+    Sensor(225, "Battery 1 BMS software version", ""),
+    Sensor(226, "Battery 1 BMS rated AH", "Ah"),
+    Sensor(227, "Battery 1 BMS hardware version", ""),
+    EnumSensor(230, "Battery 2 Manufacturer", options=hv_battery_manufacturers),
+    Sensor(241, "Battery 2 BMS charging voltage", VOLT, 0.1),
+    Sensor(242, "Battery 2 BMS discharging voltage", VOLT, 0.1),
+    Sensor(243, "Battery 2 BMS charging current limit", AMPS),
+    Sensor(244, "Battery 2 BMS discharging current limit", AMPS),
+    Sensor(245, "Battery 2 BMS SOC", "%"),
+    Sensor(246, "Battery 2 BMS voltage", VOLT, 0.1),
+    Sensor(247, "Battery 2 BMS current", AMPS),
+    TempSensor(248, "Battery 2 BMS temperature", CELSIUS, 0.1),
+    Sensor(249, "Battery 2 BMS max charge current limit", AMPS),
+    Sensor(250, "Battery 2 BMS max discharge current limit", AMPS),
+    Sensor(251, "Battery 2 BMS alarm flag", ""),
+    Sensor(252, "Battery 2 BMS fault flag", ""),
+    SelectRWSensor(253, "Battery 2 BMS other flag - NULL", options={0: "NULL"}, bitmask=0x01),
+    SelectRWSensor(253, "Battery 2 BMS other flag - Battery 1 Force charge", options={0: "off", 1: "on"}, bitmask=0x02),
+    SelectRWSensor(253, "Battery 2 BMS other flag - Battery 2 Force charge", options={0: "off", 1: "on"}, bitmask=0x04),
+    SelectRWSensor(253, "Battery 2 BMS other flag - Battery 2 Sleep", options={0: "off", 1: "on"}, bitmask=0x08),
+    Sensor(254, "Battery 2 BMS type", ""),
+    Sensor(255, "Battery 2 BMS SOH", "%"),
+    Sensor(256, "Battery 2 BMS software version", ""),
+    Sensor(257, "Battery 2 BMS rated AH", "Ah"),
+    Sensor(258, "Battery 2 BMS hardware version", ""),
+    
+)
+
+#################
+# Advanced Inverter Software Configuration Settings
+#################
+SENSORS += (
+    ### no idea why there is a "no work" option, but it's in the spec and for me, beep is set to No Work when I disabled in on the inverter...
+    SelectRWSensor(228, "Time synchronization", options={0: "No work", 1: "No work", 2: "Disable", 3: "Enable"}, bitmask=0x03),
+    SelectRWSensor(228, "Beep", options={0: "No work", 4: "No work", 8: "Disable", 12: "Enable"}, bitmask=0x0C), # mine was set to 4 after I disabled it on the inverter
+    SelectRWSensor(228, "AM/PM", options={0: "No work", 16: "No work", 32: "Disable", 48: "Enable"}, bitmask=0x30), #I guess this is 24 hour or AM/PM time display
+    SelectRWSensor(228, "Auto dim", options={0: "No work", 64: "No work", 128: "Disable", 192: "Enable"}, bitmask=0xC0),
+    SelectRWSensor(228, "Allow Remote", options={0: "No work", 16384: "No work", 32768: "Disable", 49152: "Enable"}, bitmask=0xC000), # mine was set to 32768 from factory (a "no work" option)
```

Comment:
I have to admit, I'm not a fan of having only a bitmask and some meaningless value. It works but its ugly. @kellerza what do you think, should we implement shifting or use it like done here?

Reply:
You can use `2<<13` or even hex `0x4000`

Reply:
Sure, but they are still hard to compare with the modbus manual where eg register 228 has 2 bit wide options with the values 00, 01, 10, 11. Where 10 = disable and 11 = enable and other 2 unused. There are more registers with the same pattern. Anyway, I'm not against it.

Reply:
You can also combine bniiary + shift

```python
>>> 0b10<<13
16384
>>> 2<<13
16384
```

Reply:
if you use the same shift value for the mask and values it should be clearer

```python
>>> bin(0xc000)
'0b1100000000000000'
>>> bin(0b11<<14)
'0b1100000000000000'
```

Reply:
So the question is, do you prefer to have 49152 or 0xC000 or 0x11<<14 in the code?

Reply:
`0x11<<14` is different than `0b11<<14`

but if you have `<<14` for mask+values in the same line it should be fairly clear

Reply:
Yes, my mistake, I meant 0b11<<14. And having the same shift for mask and value makes it readable enough. I have a patch in the queue waiting for LV test which needs the same.

Reply:
agree it's nice to have it more readable, I'll make the change

Reply:
done